### DISCUSSION TITLE
Windows update KB5039884 breaks oshi.jar  6.6.1 -> java.lang.ClassNotFoundException: com.sun.jna.platform.win32.Psapi$PERFORMANCE_INFORMATION

### DISCUSSION THREAD

**Discussion by AitorGS** [oshi/oshi#2688] [Q&A]
Hello

Last friday we deployed a massive windows update KB5039884 (details https://support.microsoft.com/en-us/topic/july-9-2024-kb5039884-cumulative-update-for-net-framework-3-5-and-4-8-for-windows-10-version-21h2-and-windows-10-version-22h2-2bd4310c-948e-4808-ab4e-a2e0e3bcf5bd)

Since that, all our spring boot applications started to throw exceptions when trying to invoke the SystemInfo.HardwareAbstractionLayer.getMemory with java 1.8:
java.lang.ClassNotFoundException: com.sun.jna.platform.win32.Psapi$PERFORMANCE_INFORMATION

Luckily we found an old application that was working and realized it was using oshi 6.5.0. This version does not seem to be affected by the issue so we are now redeploying hundred of applications to this version

Just wondering if this was really bad luck or what is the story behind this issue?

**Comment by dbwiddis** [comment]
He @AitorGS thanks for reporting this.

OSHI hasn't changed anything about its Windows memory code since well before 6.5.0, and I don't see any change in [Windows Documentation](https://learn.microsoft.com/en-us/windows/win32/api/psapi/ns-psapi-performance_information) for that structure, which has been around since Windows XP.

Note that the exception you're seeing is from JNA, but the current (5.14.0) version of that was included in OSHI 6.5.0, so I don't think anything changed there, either.

This type of error is [very common](https://github.com/oshi/oshi/blob/master/src/site/markdown/FAQ.md#how-do-i-resolve-jna-noclassdeffounderror-or-nosuchmethoderror-issues) when there are dependency resolution conflicts where you may be getting an older version of either `jna` or more likely `jna-platform` inherited from another dependency.  

Were there any other changes with your dependencies or Spring Boot version or any other configurations other than the Windows update?
